### PR TITLE
Add types for react-native-draggable-flatlist

### PR DESCRIPTION
**Pull request by ibarrae**

### CODE REVIEW

**Comment by ibarrae** [comment]
When you have some time, could you please take a look @CristhianMotoche

File: types/react-native-draggable-flatlist/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  /**
+   * Function that calls move when the row should become active (in an onPress, onLongPress, etc). Calls moveEnd when the gesture is complete (in onPressOut).
+   */
+  renderItem: (info: RenderItemInfo<Item>) => React.ReactElement<any> | null;
+}
+
+declare class DraggableFlatList<Item> extends Component<DraggableFlatListProps<Item>> {
+  constructor(props: DraggableFlatListProps<Item>);
+ }
```
Comment (CristhianMotoche) [inline types/react-native-draggable-flatlist/index.d.ts]: @ibarrae Not big deal. But, could you please remove that single space?

File: types/react-native-draggable-flatlist/react-native-draggable-flatlist-tests.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  render() {
+    const { data } = this.state;
+    return (
+      <DraggableFlatList
+        data={data}
+        renderItem={this.renderItem}
+        keyExtractor={(_item, index) => `draggable-item-${index}`}
+        scrollPercent={10}
+        onMoveEnd={({ data }) => this.setState({ data })}
```
Comment (CristhianMotoche) [inline types/react-native-draggable-flatlist/react-native-draggable-flatlist-tests.tsx]: @ibarrae `onMoveEnd={({ data }) => this.setState({ data })} == onMoveEnd=this.setState` isn't it the same?

Comment (ibarrae) [reply types/react-native-draggable-flatlist/react-native-draggable-flatlist-tests.tsx]: No, you can see [here](https://github.com/stackbuilders/DefinitelyTyped/pull/2/files/ed9a170d78a529848351a67759ed505f3a705742#diff-3db1e04339a2b0ec648c91064b0daa23R19) that `onMoveEnd` is a bit more complex object and by doing `onMoveEnd={this.setState}` we would be adding other items not in the state (at least in this example).

**Pull request merged**